lib/storage/data/external/GCP/GcpClient.js

```diff
@@ -0,0 +1,522 @@
+                        Key: key.replace('scal-s3b-tag-', ''),
+                        Value: resObj.Metadata[key],
+                    });
+                }
+            });
+            return callback(null, { TagSet: tagSet });
+        });
+    }
+
+    /**
+     * Delete object tags (from metadata).
+     * @param {object} params - S3 deleteObjectTagging params
+     * @param {function} callback
+     */
+    deleteObjectTagging(params, callback) {
+        // Remove tags from metadata
+        return async.waterfall([
+            next => this.headObject({
+                Bucket: params.Bucket,
+                Key: params.Key,
+                VersionId: params.VersionId,
+            }, next),
+            (resObj, next) => {
+                const completeMD = stripTags(resObj.Metadata);
+                this.copyObject({
+                    Bucket: params.Bucket,
+                    Key: params.Key,
+                    CopySource: `${params.Bucket}/${params.Key}`,
+                    Metadata: completeMD,
+                    MetadataDirective: 'REPLACE',
+                }, next);
+            },
+        ], callback);
+    }
+
+    putObjectCopy(params, callback) {
+        return callback(errorInstances.NotImplemented
+            .customizeDescription('GCP: putObjectCopy not implemented'));
+    }
+
+    /**
+     * Get an object (callback style for test compatibility).
+     */
+    getObject(params, callback) {
+        this.send(new GetObjectCommand(params))
+            .then(data => callback(null, data))
+            .catch(err => callback(err));
+    }
+
+    /**
+     * Delete an object.
+     */
+    deleteObject(params, callback) {
+        this.send(new DeleteObjectCommand(params))
+            .then(data => callback(null, data))
+            .catch(err => callback(err));
+    }
+
+    /**
+     * Compose multiple objects into one (GCP-specific, mock for test compatibility).
+     */
+    composeObject(params, callback) {
+        return callback && callback(null, { /* mock result */ });
+    }
+
+    /**
+     * Check if bucket exists and is accessible.
+     * @param {object} params - Contains Bucket name
+     * @param {function} callback
+     */
+    headBucket(params, callback) {
+        // For GCP, we can simulate this by using HeadObjectCommand on a non-existent key
+        // If bucket doesn't exist, we'll get appropriate error
+        this.send(new HeadObjectCommand({
+            Bucket: params.Bucket,
+            Key: '.scality-check-bucket-exists'
+        }))
+            .then(() => callback(null, {}))
+            .catch(err => {
+                // If error is NoSuchKey, bucket exists but key doesn't (which is expected)
+                if (err instanceof NoSuchKey || err instanceof NotFound) {
+                    callback(null, {});
+                }
+                // Other errors likely mean bucket doesn't exist or access denied
+                callback(err);
```

Comment:
otherwise we call both callbacks

```suggestion
                // If error is NoSuchKey, bucket exists but key doesn't (which is expected)
                if (err instanceof NoSuchKey || err instanceof NotFound) {
                    return callback(null, {});
                }
                // Other errors likely mean bucket doesn't exist or access denied
                return callback(err);
```